src/pages/Components/CardsList.js/CardsList.jsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react'
+import styles from "./CardsList.module.css"
+import { useAPI } from '../../../contexts/APIContext/APIContext'
+import { onlyItemsWithLayeringArt, onlyItemsWithQuilling, onlyItemsWithShadowBox } from '../../../redux/product-list-reducer';
+import { useNavigate } from 'react-router-dom';
+import { useProduct } from '../../../contexts/ProductListingContext/ProductListingContext';
+
+
+export const CardsList = () => {
+    
+    const { categories, products } = useAPI();
+    const navigate = useNavigate();
+    const { dispatch: storeDispatch } = useProduct();
+
+    const loadCategory = (type, payload) => {
+        switch(type){
+            case "QUILLING": {
+                navigate("/store")
+                return storeDispatch(onlyItemsWithQuilling(payload))
+            }
+            
+            case "LAYERING ART": {
+                navigate("/store")
+                return storeDispatch(onlyItemsWithLayeringArt(payload))
+            }
+            
+            case "SHADOW BOX": {
+                navigate("/store")
+                return storeDispatch(onlyItemsWithShadowBox(payload))
```

Comment:
try saving the action types as variables

Reply:
These are simply the values of category names for if-else conditioning. Action types contain underscore instead of spaces so I couldn't use them here.